test/helpers/helpers.go

```diff
@@ -542,3 +548,38 @@ func CheckSubscriptionNotUpdated(client test.FrameworkClient, name, namespace st
 	}
 	return nil
 }
+
+// EnsureClusterOperatorIsAvailable will make a single attempt to add the ClusterOperator
+// to the FrameworkScheme. If an error is encountered in this first call, it will
+// be returned.
+// Subsequent calls will always return whether or not the ClusterOperator was added
+// to the FrameworkScheme and nil.
+// The boolean returned by this method can be used to identify if the tests are
+// being ran on an OpenShift cluster.
+func EnsureClusterOperatorIsAvailable() (bool, error) {
```

Comment:
I suggest not doing things this way. This should be a function that is run once upon initialization and then you have another function that indicates availability. That was the pattern you were following before and I am not sure why you went away from it. Just my 2 bits.

Reply:
@kevinrizza and I had discussed this offline when he reviewed my code. IIRC, his argument was that we shouldn't impose ordering / dependencies on the order that we call functions. That by merging these two functions into a singular one, we guarantee that changes to the first function will not affect the behavior of the second function.

Reply:
I will bring this up with @kevinrizza when he returns from PTO so he may weigh in.

Reply:
That is a fair argument but the way you are using the function is inconsistent and uses the author's knowledge about execution order. Otherwise you would always be checking for error in all places. By not doing so you are in fact introducing ordering indirectly.

Reply:
That is a fair point as well @aravindhp.